lib/gyp/generator/ninja/index.js

```diff
@@ -140,6 +143,8 @@ Ninja.prototype.expand = function expand(p, productDir) {
     p = p.replace(/\$!INTERMEDIATE_DIR/g, intDir);
   }
 
+  // First one is in order to outsmart silly GYP file writers
+  p = p.replace(/(\.\.[\\\/])+\$\|CONFIGURATION_NAME/g, productDir);
```

Comment:
Alright, so what was the exact problem? Which path did it get here, and what was the right path to expect?